### PR TITLE
Docs: Initial

### PR DESCRIPTION
**Pull request by barelyhuman**
Addition of initial documentation 

Expected contents 
- [x] Installation Guide 
- [x] Usage 
   - [x] Basic Usage 
   - [x] Usage with something like Yup 
   - [x] Usage with self defined validation functions (redirected to examples instead)
   - [x] Form Level Validation 
   - [x] Guidelines on how to decide between what goes in form level validation
- [x] API overview

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,3 +6,218 @@
...
+atomWithValidate(atomGroup, validator);
+```
+
+| param       | description                                        | default     |
+| ----------- | -------------------------------------------------- | ----------- |
+| `atomGroup` | an object of `atomsWithValidate` atoms             | `undefined` |
+|             | the keys of the atomGroup are used as names/labels |             |
+|             | when passed to the `validator` function            |             |
+| `validator` | function returning void or `throw`s an error       | `undefined` |
+|             | receives the atomGroup's values with the same keys |             |
```
Comment (dai-shi) [inline README.md]: Would you like to describe about async validator? Same for options.validate in atomWithValidate.

**Review by dai-shi — APPROVED** [review]
Other than that, it looks good.

File: README.md (hunk 1 above)
Comment (barelyhuman) [reply README.md]: Sure, I'll add in a bit more detail about the validators

Comment (barelyhuman) [reply README.md]: Updated

Comment (dai-shi) [reply README.md]: feel free to merge it.

**Pull request merged**